layers/cmd/pipeline.go

```diff
@@ -0,0 +1,50 @@
+package cmd
+
+import (
+	"io"
+	"path/filepath"
+
+	"github.com/cloudprivacylabs/lsa/pkg/ls"
+	"github.com/cloudprivacylabs/opencypher/graph"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	ingestCmd.AddCommand(pipelineCmd)
+	pipelineCmd.Flags().String("file", "", "Pipeline build file")
+	pipelineCmd.Flags().String("inputFile", "", "User provided input file")
+}
+
+var pipelineCmd = &cobra.Command{
+	Use:   "pipeline",
+	Short: "build pipeline JSON",
+	Args:  cobra.MaximumNArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		inputFile, err := cmd.Flags().GetString("inputFile")
+		if err != nil {
+			failErr(err)
+		}
+		file, err := cmd.Flags().GetString("file")
+		if err != nil {
+			failErr(err)
+		}
+		format := filepath.Ext(file)
```

Comment:
No need for format here. Pipeline should process the steps here instead.

Let's do this:
type PipelineContext struct {
  ...
   currentStep int
   steps []Step
}

Then:

func (ctx *PipelineContext) Next() error {
  ctx.currentStep++
   if ctx.currentStep>= len(ctx.steps){ 
     return nil
  }
  return ctx.steps[ctx.currentStep].Run(ctx)
}

This way, each step calls ctx.Next) when then process input.